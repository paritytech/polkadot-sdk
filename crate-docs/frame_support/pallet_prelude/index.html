<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Prelude to be used alongside pallet macro, for ease of use."><title>frame_support::pallet_prelude - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../static.files/rustdoc-cb6f1f67f1bcd037.css" id="mainThemeStyle"><meta name="rustdoc-vars" data-root-path="../../" data-static-root-path="../../static.files/" data-current-crate="frame_support" data-themes="" data-resource-suffix="" data-rustdoc-version="1.73.0 (cc66ad468 2023-10-03)" data-channel="1.73.0" data-search-js="search-6dfdfced5eff6596.js" data-settings-js="settings-de11bff964e9d4e5.js" data-settings-css="settings-8c76f75bfb6bd192.css" data-theme-light-css="light-1596385f77d47ef2.css" data-theme-dark-css="dark-0a43001d3fc2282c.css" data-theme-ayu-css="ayu-fd19013d6ce078bf.css" ><script src="../../static.files/storage-db41da1a38ea3cb8.js"></script><script defer src="../../static.files/main-0795b7d26be81095.js"></script><noscript><link rel="stylesheet" media="(prefers-color-scheme:light)" href="../../static.files/light-1596385f77d47ef2.css"><link rel="stylesheet" media="(prefers-color-scheme:dark)" href="../../static.files/dark-0a43001d3fc2282c.css"><link rel="stylesheet" href="../../static.files/noscript-cffde32267a19fd6.css"></noscript><link rel="alternate icon" type="image/png" href="../../static.files/favicon-16x16-8b506e7a72182f1c.png"><link rel="alternate icon" type="image/png" href="../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../static.files/favicon-2c020d218678b618.svg"><script async defer src="https://apisa.parity.io/latest.js"></script><noscript><img src="https://apisa.parity.io/latest.js" alt="" referrerpolicy="no-referrer-when-downgrade" /></noscript></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="logo-container" href="../../frame_support/index.html"><img class="rust-logo" src="../../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2></h2></nav><nav class="sidebar"><a class="logo-container" href="../../frame_support/index.html"><img class="rust-logo" src="../../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2 class="location"><a href="#">Module pallet_prelude</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#reexports">Re-exports</a></li><li><a href="#macros">Macros</a></li><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#constants">Constants</a></li><li><a href="#traits">Traits</a></li><li><a href="#types">Type Definitions</a></li><li><a href="#attributes">Attribute Macros</a></li><li><a href="#derives">Derive Macros</a></li></ul></section></div></nav><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><a href="../../help.html">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../index.html">frame_support</a>::<wbr><a class="mod" href="#">pallet_prelude</a><button id="copy-path" title="Copy item path to clipboard"><img src="../../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="src" href="../../src/frame_support/lib.rs.html#803">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Prelude to be used alongside pallet macro, for ease of use.</p>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2><ul class="item-table"><li><div class="item-name" id="reexport.DispatchClass"><code>pub use crate::dispatch::<a class="enum" href="../dispatch/enum.DispatchClass.html" title="enum frame_support::dispatch::DispatchClass">DispatchClass</a>;</code></div></li><li><div class="item-name" id="reexport.DispatchResult"><code>pub use crate::dispatch::<a class="type" href="../dispatch/type.DispatchResult.html" title="type frame_support::dispatch::DispatchResult">DispatchResult</a>;</code></div></li><li><div class="item-name" id="reexport.DispatchResultWithPostInfo"><code>pub use crate::dispatch::<a class="type" href="../dispatch/type.DispatchResultWithPostInfo.html" title="type frame_support::dispatch::DispatchResultWithPostInfo">DispatchResultWithPostInfo</a>;</code></div></li><li><div class="item-name" id="reexport.Parameter"><code>pub use crate::dispatch::<a class="trait" href="../dispatch/trait.Parameter.html" title="trait frame_support::dispatch::Parameter">Parameter</a>;</code></div></li><li><div class="item-name" id="reexport.Pays"><code>pub use crate::dispatch::<a class="enum" href="../dispatch/enum.Pays.html" title="enum frame_support::dispatch::Pays">Pays</a>;</code></div></li><li><div class="item-name" id="reexport.ProvideInherent"><code>pub use crate::inherent::<a class="trait" href="../inherent/trait.ProvideInherent.html" title="trait frame_support::inherent::ProvideInherent">ProvideInherent</a>;</code></div></li><li><div class="item-name" id="reexport.storage"><code>pub use crate::<a class="mod" href="../storage/index.html" title="mod frame_support::storage">storage</a>;</code></div></li><li><div class="item-name" id="reexport.CountedStorageMap"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.CountedStorageMap.html" title="struct frame_support::storage::types::CountedStorageMap">CountedStorageMap</a>;</code></div></li><li><div class="item-name" id="reexport.CountedStorageNMap"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.CountedStorageNMap.html" title="struct frame_support::storage::types::CountedStorageNMap">CountedStorageNMap</a>;</code></div></li><li><div class="item-name" id="reexport.NMapKey"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.Key.html" title="struct frame_support::storage::types::Key">Key</a> as NMapKey;</code></div></li><li><div class="item-name" id="reexport.OptionQuery"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.OptionQuery.html" title="struct frame_support::storage::types::OptionQuery">OptionQuery</a>;</code></div></li><li><div class="item-name" id="reexport.ResultQuery"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.ResultQuery.html" title="struct frame_support::storage::types::ResultQuery">ResultQuery</a>;</code></div></li><li><div class="item-name" id="reexport.StorageDoubleMap"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.StorageDoubleMap.html" title="struct frame_support::storage::types::StorageDoubleMap">StorageDoubleMap</a>;</code></div></li><li><div class="item-name" id="reexport.StorageMap"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.StorageMap.html" title="struct frame_support::storage::types::StorageMap">StorageMap</a>;</code></div></li><li><div class="item-name" id="reexport.StorageNMap"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.StorageNMap.html" title="struct frame_support::storage::types::StorageNMap">StorageNMap</a>;</code></div></li><li><div class="item-name" id="reexport.StorageValue"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.StorageValue.html" title="struct frame_support::storage::types::StorageValue">StorageValue</a>;</code></div></li><li><div class="item-name" id="reexport.ValueQuery"><code>pub use crate::storage::types::<a class="struct" href="../storage/types/struct.ValueQuery.html" title="struct frame_support::storage::types::ValueQuery">ValueQuery</a>;</code></div></li><li><div class="item-name" id="reexport.StorageList"><code>pub use crate::storage::<a class="trait" href="../storage/trait.StorageList.html" title="trait frame_support::storage::StorageList">StorageList</a>;</code></div></li><li><div class="item-name" id="reexport.BuildGenesisConfig"><code>pub use crate::traits::<a class="trait" href="../traits/trait.BuildGenesisConfig.html" title="trait frame_support::traits::BuildGenesisConfig">BuildGenesisConfig</a>;</code></div></li><li><div class="item-name" id="reexport.EnsureOrigin"><code>pub use crate::traits::<a class="trait" href="../traits/trait.EnsureOrigin.html" title="trait frame_support::traits::EnsureOrigin">EnsureOrigin</a>;</code></div></li><li><div class="item-name" id="reexport.GetStorageVersion"><code>pub use crate::traits::<a class="trait" href="../traits/trait.GetStorageVersion.html" title="trait frame_support::traits::GetStorageVersion">GetStorageVersion</a>;</code></div></li><li><div class="item-name" id="reexport.Hooks"><code>pub use crate::traits::<a class="trait" href="../traits/trait.Hooks.html" title="trait frame_support::traits::Hooks">Hooks</a>;</code></div></li><li><div class="item-name" id="reexport.IsType"><code>pub use crate::traits::<a class="trait" href="../traits/trait.IsType.html" title="trait frame_support::traits::IsType">IsType</a>;</code></div></li><li><div class="item-name" id="reexport.PalletInfoAccess"><code>pub use crate::traits::<a class="trait" href="../traits/trait.PalletInfoAccess.html" title="trait frame_support::traits::PalletInfoAccess">PalletInfoAccess</a>;</code></div></li><li><div class="item-name" id="reexport.StorageInfoTrait"><code>pub use crate::traits::<a class="trait" href="../traits/trait.StorageInfoTrait.html" title="trait frame_support::traits::StorageInfoTrait">StorageInfoTrait</a>;</code></div></li><li><div class="item-name" id="reexport.StorageVersion"><code>pub use crate::traits::<a class="struct" href="../traits/struct.StorageVersion.html" title="struct frame_support::traits::StorageVersion">StorageVersion</a>;</code></div></li><li><div class="item-name" id="reexport.Blake2_128"><code>pub use crate::<a class="struct" href="../struct.Blake2_128.html" title="struct frame_support::Blake2_128">Blake2_128</a>;</code></div></li><li><div class="item-name" id="reexport.Blake2_128Concat"><code>pub use crate::<a class="struct" href="../struct.Blake2_128Concat.html" title="struct frame_support::Blake2_128Concat">Blake2_128Concat</a>;</code></div></li><li><div class="item-name" id="reexport.Blake2_256"><code>pub use crate::<a class="struct" href="../struct.Blake2_256.html" title="struct frame_support::Blake2_256">Blake2_256</a>;</code></div></li><li><div class="item-name" id="reexport.Identity"><code>pub use crate::<a class="struct" href="../struct.Identity.html" title="struct frame_support::Identity">Identity</a>;</code></div></li><li><div class="item-name" id="reexport.Twox128"><code>pub use crate::<a class="struct" href="../struct.Twox128.html" title="struct frame_support::Twox128">Twox128</a>;</code></div></li><li><div class="item-name" id="reexport.Twox256"><code>pub use crate::<a class="struct" href="../struct.Twox256.html" title="struct frame_support::Twox256">Twox256</a>;</code></div></li><li><div class="item-name" id="reexport.Twox64Concat"><code>pub use crate::<a class="struct" href="../struct.Twox64Concat.html" title="struct frame_support::Twox64Concat">Twox64Concat</a>;</code></div></li><li><div class="item-name"><code>pub use frame_support::<a class="mod" href="../pallet_macros/index.html" title="mod frame_support::pallet_macros">pallet_macros</a>::*;</code></div></li></ul><h2 id="macros" class="small-section-header"><a href="#macros">Macros</a></h2><ul class="item-table"><li><div class="item-name"><a class="macro" href="macro.defensive.html" title="macro frame_support::pallet_prelude::defensive">defensive</a></div><div class="desc docblock-short">Generic function to mark an execution path as ONLY defensive.</div></li><li><div class="item-name"><a class="macro" href="macro.defensive_assert.html" title="macro frame_support::pallet_prelude::defensive_assert">defensive_assert</a></div><div class="desc docblock-short">Trigger a defensive failure if a condition is not met.</div></li><li><div class="item-name"><a class="macro" href="macro.ensure.html" title="macro frame_support::pallet_prelude::ensure">ensure</a></div><div class="desc docblock-short">Evaluate <code>$x:expr</code> and if not true return <code>Err($y:expr)</code>.</div></li></ul><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2><ul class="item-table"><li><div class="item-name"><a class="struct" href="struct.BoundedBTreeMap.html" title="struct frame_support::pallet_prelude::BoundedBTreeMap">BoundedBTreeMap</a></div><div class="desc docblock-short">A bounded map based on a B-Tree.</div></li><li><div class="item-name"><a class="struct" href="struct.BoundedBTreeSet.html" title="struct frame_support::pallet_prelude::BoundedBTreeSet">BoundedBTreeSet</a></div><div class="desc docblock-short">A bounded set based on a B-Tree.</div></li><li><div class="item-name"><a class="struct" href="struct.BoundedVec.html" title="struct frame_support::pallet_prelude::BoundedVec">BoundedVec</a></div><div class="desc docblock-short">A bounded vector.</div></li><li><div class="item-name"><a class="struct" href="struct.ConstU32.html" title="struct frame_support::pallet_prelude::ConstU32">ConstU32</a></div><div class="desc docblock-short">Const getter for a basic type.</div></li><li><div class="item-name"><a class="struct" href="struct.GetDefault.html" title="struct frame_support::pallet_prelude::GetDefault">GetDefault</a></div><div class="desc docblock-short">Implement Get by returning Default for any type that implements Default.</div></li><li><div class="item-name"><a class="struct" href="struct.InherentData.html" title="struct frame_support::pallet_prelude::InherentData">InherentData</a></div><div class="desc docblock-short">Inherent data to include in a block.</div></li><li><div class="item-name"><a class="struct" href="struct.MakeFatalError.html" title="struct frame_support::pallet_prelude::MakeFatalError">MakeFatalError</a></div><div class="desc docblock-short">Auxiliary to make any given error resolve to <code>is_fatal_error() == true</code> for <a href="../inherent/trait.IsFatalError.html" title="trait frame_support::inherent::IsFatalError"><code>IsFatalError</code></a>.</div></li><li><div class="item-name"><a class="struct" href="struct.PhantomData.html" title="struct frame_support::pallet_prelude::PhantomData">PhantomData</a></div><div class="desc docblock-short">Zero-sized type used to mark things that “act like” they own a <code>T</code>.</div></li><li><div class="item-name"><a class="struct" href="struct.ValidTransaction.html" title="struct frame_support::pallet_prelude::ValidTransaction">ValidTransaction</a></div><div class="desc docblock-short">Information concerning a valid transaction.</div></li><li><div class="item-name"><a class="struct" href="struct.WeakBoundedVec.html" title="struct frame_support::pallet_prelude::WeakBoundedVec">WeakBoundedVec</a></div><div class="desc docblock-short">A weakly bounded vector.</div></li><li><div class="item-name"><a class="struct" href="struct.Weight.html" title="struct frame_support::pallet_prelude::Weight">Weight</a></div></li></ul><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2><ul class="item-table"><li><div class="item-name"><a class="enum" href="enum.DispatchError.html" title="enum frame_support::pallet_prelude::DispatchError">DispatchError</a></div><div class="desc docblock-short">Reason why a dispatch call failed.</div></li><li><div class="item-name"><a class="enum" href="enum.InvalidTransaction.html" title="enum frame_support::pallet_prelude::InvalidTransaction">InvalidTransaction</a></div><div class="desc docblock-short">An invalid transaction validity.</div></li><li><div class="item-name"><a class="enum" href="enum.TransactionSource.html" title="enum frame_support::pallet_prelude::TransactionSource">TransactionSource</a></div><div class="desc docblock-short">The source of the transaction.</div></li><li><div class="item-name"><a class="enum" href="enum.TransactionValidityError.html" title="enum frame_support::pallet_prelude::TransactionValidityError">TransactionValidityError</a></div><div class="desc docblock-short">Errors that can occur while checking the validity of a transaction.</div></li><li><div class="item-name"><a class="enum" href="enum.UnknownTransaction.html" title="enum frame_support::pallet_prelude::UnknownTransaction">UnknownTransaction</a></div><div class="desc docblock-short">An unknown transaction validity.</div></li></ul><h2 id="constants" class="small-section-header"><a href="#constants">Constants</a></h2><ul class="item-table"><li><div class="item-name"><a class="constant" href="constant.MAX_MODULE_ERROR_ENCODED_SIZE.html" title="constant frame_support::pallet_prelude::MAX_MODULE_ERROR_ENCODED_SIZE">MAX_MODULE_ERROR_ENCODED_SIZE</a></div><div class="desc docblock-short">The number of bytes of the module-specific <code>error</code> field defined in <a href="../../sp_runtime/struct.ModuleError.html" title="struct sp_runtime::ModuleError"><code>ModuleError</code></a>.
In FRAME, this is the maximum encoded size of a pallet error type.</div></li></ul><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2><ul class="item-table"><li><div class="item-name"><a class="trait" href="trait.Decode.html" title="trait frame_support::pallet_prelude::Decode">Decode</a></div><div class="desc docblock-short">Trait that allows zero-copy read of value-references from slices in LE format.</div></li><li><div class="item-name"><a class="trait" href="trait.Encode.html" title="trait frame_support::pallet_prelude::Encode">Encode</a></div><div class="desc docblock-short">Trait that allows zero-copy write of value-references to slices in LE format.</div></li><li><div class="item-name"><a class="trait" href="trait.Get.html" title="trait frame_support::pallet_prelude::Get">Get</a></div><div class="desc docblock-short">A trait for querying a single value from a type.</div></li><li><div class="item-name"><a class="trait" href="trait.MaxEncodedLen.html" title="trait frame_support::pallet_prelude::MaxEncodedLen">MaxEncodedLen</a></div><div class="desc docblock-short">Items implementing <code>MaxEncodedLen</code> have a statically known maximum encoded size.</div></li><li><div class="item-name"><a class="trait" href="trait.MaybeSerializeDeserialize.html" title="trait frame_support::pallet_prelude::MaybeSerializeDeserialize">MaybeSerializeDeserialize</a></div><div class="desc docblock-short">A type that implements Serialize, DeserializeOwned and Debug when in std environment or serde feature is activated.</div></li><li><div class="item-name"><a class="trait" href="trait.Member.html" title="trait frame_support::pallet_prelude::Member">Member</a></div><div class="desc docblock-short">A type that can be used in runtime structures.</div></li><li><div class="item-name"><a class="trait" href="trait.TypeInfo.html" title="trait frame_support::pallet_prelude::TypeInfo">TypeInfo</a></div><div class="desc docblock-short">Implementors return their meta type information.</div></li><li><div class="item-name"><a class="trait" href="trait.TypedGet.html" title="trait frame_support::pallet_prelude::TypedGet">TypedGet</a></div><div class="desc docblock-short">A trait for querying a single value from a type defined in the trait.</div></li><li><div class="item-name"><a class="trait" href="trait.ValidateUnsigned.html" title="trait frame_support::pallet_prelude::ValidateUnsigned">ValidateUnsigned</a></div><div class="desc docblock-short">Provide validation for unsigned extrinsics.</div></li></ul><h2 id="types" class="small-section-header"><a href="#types">Type Definitions</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.InherentIdentifier.html" title="type frame_support::pallet_prelude::InherentIdentifier">InherentIdentifier</a></div><div class="desc docblock-short">An identifier for an inherent.</div></li><li><div class="item-name"><a class="type" href="type.TransactionLongevity.html" title="type frame_support::pallet_prelude::TransactionLongevity">TransactionLongevity</a></div><div class="desc docblock-short">Minimum number of blocks a transaction will remain valid for.
<code>TransactionLongevity::max_value()</code> means “forever”.</div></li><li><div class="item-name"><a class="type" href="type.TransactionPriority.html" title="type frame_support::pallet_prelude::TransactionPriority">TransactionPriority</a></div><div class="desc docblock-short">Priority for a transaction. Additive. Higher is better.</div></li><li><div class="item-name"><a class="type" href="type.TransactionTag.html" title="type frame_support::pallet_prelude::TransactionTag">TransactionTag</a></div><div class="desc docblock-short">Tag for a transaction. No two transactions with the same tag should be placed on-chain.</div></li><li><div class="item-name"><a class="type" href="type.TransactionValidity.html" title="type frame_support::pallet_prelude::TransactionValidity">TransactionValidity</a></div><div class="desc docblock-short">Information on a transaction’s validity and, if valid, on how it relates to other transactions.</div></li></ul><h2 id="attributes" class="small-section-header"><a href="#attributes">Attribute Macros</a></h2><ul class="item-table"><li><div class="item-name"><a class="attr" href="attr.inject_runtime_type.html" title="attr frame_support::pallet_prelude::inject_runtime_type">inject_runtime_type</a></div><div class="desc docblock-short">The optional attribute <code>#[inject_runtime_type]</code> can be attached to <code>RuntimeCall</code>,
<code>RuntimeEvent</code>, <code>RuntimeOrigin</code> or <code>PalletInfo</code> in an impl statement that has
<code>#[register_default_impl]</code> attached to indicate that this item is generated by
<code>construct_runtime</code>.</div></li><li><div class="item-name"><a class="attr" href="attr.register_default_impl.html" title="attr frame_support::pallet_prelude::register_default_impl">register_default_impl</a></div><div class="desc docblock-short">Attach this attribute to an impl statement that you want to use with
<a href="../attr.derive_impl.html" title="attr frame_support::derive_impl"><code>#[derive_impl(..)]</code></a>.</div></li></ul><h2 id="derives" class="small-section-header"><a href="#derives">Derive Macros</a></h2><ul class="item-table"><li><div class="item-name"><a class="derive" href="derive.CloneNoBound.html" title="derive frame_support::pallet_prelude::CloneNoBound">CloneNoBound</a></div><div class="desc docblock-short">Derive <a href="https://doc.rust-lang.org/1.73.0/core/clone/trait.Clone.html" title="trait core::clone::Clone"><code>Clone</code></a> but do not bound any generic. Docs are at <code>frame_support::CloneNoBound</code>.</div></li><li><div class="item-name"><a class="derive" href="derive.DebugNoBound.html" title="derive frame_support::pallet_prelude::DebugNoBound">DebugNoBound</a></div><div class="desc docblock-short">Derive <a href="https://doc.rust-lang.org/1.73.0/core/fmt/macros/derive.Debug.html" title="derive core::fmt::macros::Debug"><code>Debug</code></a> but do not bound any generics. Docs are at <code>frame_support::DebugNoBound</code>.</div></li><li><div class="item-name"><a class="derive" href="derive.Decode.html" title="derive frame_support::pallet_prelude::Decode">Decode</a></div><div class="desc docblock-short">Derive <code>parity_scale_codec::Decode</code> and for struct and enum.</div></li><li><div class="item-name"><a class="derive" href="derive.Encode.html" title="derive frame_support::pallet_prelude::Encode">Encode</a></div><div class="desc docblock-short">Derive <code>parity_scale_codec::Encode</code> and <code>parity_scale_codec::EncodeLike</code> for struct and enum.</div></li><li><div class="item-name"><a class="derive" href="derive.EqNoBound.html" title="derive frame_support::pallet_prelude::EqNoBound">EqNoBound</a></div><div class="desc docblock-short">derive Eq but do no bound any generic. Docs are at <code>frame_support::EqNoBound</code>.</div></li><li><div class="item-name"><a class="derive" href="derive.MaxEncodedLen.html" title="derive frame_support::pallet_prelude::MaxEncodedLen">MaxEncodedLen</a></div><div class="desc docblock-short">Derive <code>parity_scale_codec::MaxEncodedLen</code> for struct and enum.</div></li><li><div class="item-name"><a class="derive" href="derive.PartialEqNoBound.html" title="derive frame_support::pallet_prelude::PartialEqNoBound">PartialEqNoBound</a></div><div class="desc docblock-short">Derive <a href="https://doc.rust-lang.org/1.73.0/core/cmp/trait.PartialEq.html" title="trait core::cmp::PartialEq"><code>PartialEq</code></a> but do not bound any generic. Docs are at
<code>frame_support::PartialEqNoBound</code>.</div></li><li><div class="item-name"><a class="derive" href="derive.RuntimeDebug.html" title="derive frame_support::pallet_prelude::RuntimeDebug">RuntimeDebug</a></div></li><li><div class="item-name"><a class="derive" href="derive.RuntimeDebugNoBound.html" title="derive frame_support::pallet_prelude::RuntimeDebugNoBound">RuntimeDebugNoBound</a></div><div class="desc docblock-short">Derive [<code>Debug</code>], if <code>std</code> is enabled it uses <code>frame_support::DebugNoBound</code>, if <code>std</code> is not
enabled it just returns <code>&quot;&lt;wasm:stripped&gt;&quot;</code>.
This behaviour is useful to prevent bloating the runtime WASM blob from unneeded code.</div></li><li><div class="item-name"><a class="derive" href="derive.TypeInfo.html" title="derive frame_support::pallet_prelude::TypeInfo">TypeInfo</a></div></li></ul></section></div></main></body></html>