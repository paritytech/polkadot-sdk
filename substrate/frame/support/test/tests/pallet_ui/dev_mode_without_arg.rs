// This file is part of Substrate.

// Copyright (C) Parity Technologies (UK) Ltd.
// SPDX-License-Identifier: Apache-2.0

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// 	http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#![cfg_attr(not(feature = "std"), no_std)]

#[frame_support::pallet]
pub mod pallet {
	use frame_support::pallet_prelude::*;
	use frame_system::pallet_prelude::*;

	// The struct on which we build all of our Pallet logic.
	#[pallet::pallet]
	pub struct Pallet<T>(_);

	// Your Pallet's configuration trait, representing custom external types and interfaces.
	#[pallet::config]
	pub trait Config: frame_system::Config {}

	#[pallet::storage]
	type MyStorage<T: Config> = StorageValue<_, Vec<u8>>;

	// Your Pallet's callable functions.
	#[pallet::call]
	impl<T: Config> Pallet<T> {
		pub fn my_call(_origin: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}

	// Your Pallet's internal functions.
	impl<T: Config> Pallet<T> {}
}

fn main() {}
