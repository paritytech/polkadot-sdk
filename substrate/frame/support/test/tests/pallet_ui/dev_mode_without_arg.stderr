error: A pallet::call requires either a concrete `#[pallet::weight($expr)]` or an
                               inherited weight from the `#[pallet:call(weight($type))]` attribute, but
                               none were given.
  --> tests/pallet_ui/dev_mode_without_arg.rs:39:7
   |
39 |         pub fn my_call(_origin: OriginFor<T>) -> DispatchResult {
   |             ^^
